plume-models/src/users.rs

```diff
@@ -320,13 +320,43 @@ impl User {
             .load::<User>(conn)
             .map_err(Error::from)
     }
-
     pub fn outbox(&self, conn: &Connection) -> Result<ActivityStream<OrderedCollection>> {
-        let acts = self.get_activities(conn)?;
-        let n_acts = acts.len();
         let mut coll = OrderedCollection::default();
+        let first = ap_url(&format!("{}?page=1", &self.outbox_url));
+        let last = ap_url(&format!(
+            "{}?page={}",
+            &self.outbox_url,
+            self.get_activities_count(&conn) / ITEMS_PER_PAGE as i64
+        ));
+        coll.collection_props.set_first_link(Id::new(first))?;
+        coll.collection_props.set_last_link(Id::new(last))?;
+        coll.collection_props
+            .set_total_items_u64(self.get_activities_count(&conn) as u64)?;
+        Ok(ActivityStream::new(coll))
+    }
+    pub fn outbox_page(
+        &self,
+        conn: &Connection,
+        (min, max): (i32, i32),
+    ) -> Result<ActivityStream<OrderedCollectionPage>> {
+        let acts = self.get_activities_page(conn, (min, max))?;
+        let mut coll = OrderedCollectionPage::default();
         coll.collection_props.items = serde_json::to_value(acts)?;
-        coll.collection_props.set_total_items_u64(n_acts as u64)?;
+        //coll.collection_props.set_total_items_u64(n_acts as u64)?;
+        coll.collection_page_props
+            .set_next_link(Id::new(ap_url(&format!(
+                "{}?page={}",
+                &self.outbox_url,
+                min / ITEMS_PER_PAGE + 1
+            ))))?;
+        coll.collection_page_props
+            .set_prev_link(Id::new(ap_url(&format!(
+                "{}?page={}",
+                &self.outbox_url,
+                min / ITEMS_PER_PAGE - 1
+            ))))?;
```

Comment:
Previous page should not be added if we are already on the first page, and a next page should probable not be added either if `get_activities_page` returned less entries than it could have.

Also I think there is an out-of-one on the page index, if we are on page n°1 (that is the first with current implementation of `outbox` above), min is going to be 0, so next page will be 0/12+1 = n°1 again, and previous would be n°-1

Reply:
That makes sense. Should be fixed now

Reply:
That makes sense.

Reply:
the part about not including previous/next when unnecessary is fixed, but I don't think you addressed the other half about page index